pkg/proto/configuration/global/global.proto

```diff
@@ -56,18 +57,95 @@ message PrometheusPushgatewayConfiguration {
   google.protobuf.Duration push_interval = 5;
 }
 
+message OpenCensusConfiguration {
+  // Export data to an OpenCensus agent. The recommended configuration
+  // is to use opentelemetry-collector and enable its opencensus-agent
+  // receiver.
+
+  // Connection address for the agent. Default is localhost:55678.
+  string address = 1;
+
+  // TLS configuration. TLS is not enabled when left unset.
+  buildbarn.configuration.tls.ClientConfiguration tls = 2;
+
+  // OpenCensus service name.
+  string service_name = 3;
+}
+
+message OpenTelemetryConfiguration {
+  // Connection address for the collector. Default is localhost:55680.
+  string address = 1;
+
+  // TLS configuration. TLS is not enabled when left unset.
+  buildbarn.configuration.tls.ClientConfiguration tls = 2;
+
+  // Resource detectors. Labels will be merged, with earlier things in
+  // this entries being preferred over later ones.
+  enum ResourceDetector {
+    ENV = 0;  // Use OTEL_RESOURCE_ATTRIBUTES environment variable
+    AWS = 1;  // Use AWS instance metadata
+    GCE = 2;  // Use GCE instance metadata
+    GKE = 3;  // Use GKE API
+  }
+  repeated ResourceDetector resource_detectors = 3;
+
+  // Queue size for buffering spans. Default is 2048.
+  int32 max_queue_size = 4;
+
+  // If true, then span Start() and End() functions will block when
+  // the queue is full. This will significantly degrade performance,
+  // but does guarantee no spans are lost, which can be useful when
+  // debugging with the sampling probability set to 1.0.
+  bool block_on_queue_full = 5;
```

Comment:
block_when_queue_is_full or block_on_full_queue?

Reply:
I went for matching the names used in otel: https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.11.0/trace?tab=doc#BatchSpanProcessorOptions

While the grammar is questionable, consistency seems better.